### DISCUSSION TITLE
Speaker Diarization

### DISCUSSION THREAD

**Discussion by samecfrank** [openai/whisper#340] [Ideas]
Is there a speaker diarization feature to whisper? If not will this be a feature in a future release? Thank you!

**Comment by turnkit** [comment]
I came to ask for this also. But I wasn’t familiar with the term you used. 

Am concerned you won’t get many upvotes only because the title is vague for most. 

For those that didn’t look up the Wikipedia definition: “Speaker diarization ([or](https://en.m.wikipedia.org/wiki/American_and_British_English_spelling_differences#-ise.2C_-ize_.28-isation.2C_-ization.29) diarization) (clarification: a human speaker is meant) is the process of partitioning an input audio stream into homogeneous segments according to the speaker identity. It can enhance the readability of an [automatic speech transcription](https://en.m.wikipedia.org/wiki/Speech_recognition) by structuring the audio stream into speaker turns and, when used together with [speaker recognition](https://en.m.wikipedia.org/wiki/Speaker_recognition) systems, by providing the speaker’s true identity.[[1]](https://en.m.wikipedia.org/wiki/Speaker_diarisation#cite_note-1) It is used to answer the question "who spoke when?"[[2]](https://en.m.wikipedia.org/wiki/Speaker_diarisation#cite_note-2) Speaker diarisation is a combination of speaker segmentation and speaker clustering. The first aims at finding speaker change points in an audio stream. The second aims at grouping together speech segments on the basis of speaker characteristics.

**Comment by Majdoddin** [comment]
Hi @samecfrank  @turnkit 
[www.lexicaps.com](http://www.lexicaps.com/) seamlessly adds diarization to Whispers transcription. No 3rd party packages.
Feel free to contact me if you need more features.

Announcement: https://github.com/openai/whisper/discussions/1537
Repo: https://github.com/Majdoddin/lexicaps
